milli/src/update/index_documents/transform.rs

```diff
@@ -148,6 +152,8 @@ impl Transform<'_, '_> {
             // We prepare the fields ids map with the documents keys.
             for (key, _value) in &document {
                 fields_ids_map.insert(&key).context("field id limit reached")?;
+
+                *fields_distribution.entry(key.to_owned()).or_default() += 1;
```

Comment:
We could do the same way as explained before: introducing a HashMap where the key is the field id and to the merge afterward. This way we reduce the number of string creations in this hot loop.